tests/test_jump_cas22.py

```diff
@@ -0,0 +1,444 @@
+    resultants, read_noise, read_pattern, n_pixels, jumps = jump_data
+    dq = np.zeros(resultants.shape, dtype=np.int32)
+
+    fits, _, _ = fit_ramps(resultants, dq, read_noise, ROMAN_READ_TIME, read_pattern, use_jump=True)
+
+    # Check that all the jumps have been located per the algorithm's constraints
+    for index, (fit, jump) in enumerate(zip(fits, jumps)):
+        # sanity check that only one jump should have been added
+        assert np.where(jump)[0].shape == (1,)
+        if index == 0:
+            # There is no way to detect a jump if it is in the very first read
+            # The very first pixel in this case has a jump in the first read
+            assert len(fit['jumps']) == 0
+            assert jump[0]  # sanity check that the jump is in the first resultant still
+            assert not np.all(jump[1:])
+
+            # Test that the correct index was recorded
+            assert len(fit['index']) == 1
+            assert fit['index'][0]['start'] == 0
+            assert fit['index'][0]['end'] == len(read_pattern) - 1
+        else:
+            # Select the single jump and check that it is recorded as a jump
+            assert np.where(jump)[0][0] in fit['jumps']
+
+            # In all cases here we have to exclude two resultants
+            assert len(fit['jumps']) == 2
+
+            # Test that all the jumps recorded are +/- 1 of the real jump
+            #    This is due to the need to exclude two resultants
+            for jump_index in fit['jumps']:
+                assert jump[jump_index] or jump[jump_index + 1] or jump[jump_index - 1]
+
+            # Test that the correct indexes are recorded
+            ramp_indicies = []
+            for ramp_index in fit["index"]:
+                # Note start/end of a ramp_index are inclusive meaning that end
+                #    is an index included in the ramp_index so the range is to end + 1
+                new_indicies = list(range(ramp_index["start"], ramp_index["end"] + 1))
+
+                # check that all the ramps are non-overlapping
+                assert set(ramp_indicies).isdisjoint(new_indicies)
+
+                ramp_indicies.extend(new_indicies)
+
+            # check that no ramp_index is a jump
+            assert set(ramp_indicies).isdisjoint(fit['jumps'])
+
+            # check that all resultant indicies are either in a ramp or listed as a jump
+            assert set(ramp_indicies).union(fit['jumps']) == set(range(len(read_pattern)))
+
+    # Check that the slopes have been estimated reasonably well
+    #   There are not that many pixels to test this against and many resultants
+    #   have been thrown out due to the jumps. Thus we only check the slope is
+    #   "fairly close" to the expected value. This is purposely a loose check
+    #   because the main purpose of this test is to verify that the jumps are
+    #   being detected correctly, above.
+    for fit in fits:
+        assert_allclose(fit['average']['slope'], FLUX, rtol=3)
```

Comment:
If you were to write this test like the chi^2 test, what tolerance would you need?  It is very reassuring when the chi^2/dof comes out close to 1.

Reply:
See last test, it has been modified to use enough points to get good statistical convergence.